### ISSUE TITLE
Question on running a session twice

### ISSUE THREAD

**Issue by StoyanVenDimitrov** [ufal/neuralmonkey#531]
Hi, 
after some time of using the property of the tf_manager to run a session number of times, I realized it might be a wrong approach for me. But not sure, so I ask for second opinion. The code is here: https://github.com/StoyanVenDimitrov/neuralmonkey/blob/with_new_decoder/neuralmonkey/trainers/mrt_trainer.py
The case is:
I use the decoder.py in sampling mode. It gives me some logits and vocabulary IDs. I run a session with fetches only for the IDs and obtain the sentences they represent and their GLEU score. Then I go back to gather the logits that these IDs stand for, running another session with fetches for the loss, where these gathered logits are part of. But this should be wrong. Because even if I have no fetches that involve logits on the first run, on the second run, because of the loss fetch, new logits are obtained in the decoder, right? I thought they aren't because no fetches for them have been ran yet. But session.run() should result in recomputing all nodes that are involved in the fetches. This means session.partial_run() is the only way. Am I right?   Thanks a lot!

Best,
Stoyan

**Comment by jindrahelcl** [comment]
Hi, sorry for not replying for a long time - did you solve the problem?

2017-09-19 16:08 GMT+02:00 StoyanVenDimitrov <notifications@github.com>:

**Comment by StoyanVenDimitrov** [comment]
Hi, 
No worries, it was rather a general TF question. 
Yes, I just ran the second session not in sample, but in train modus, using the output of the sampling as input for the second session.
